src/SoundAnalyser.js

```diff
@@ -0,0 +1,90 @@
+// @flow
+
+import * as C2lcMath from './C2lcMath';
+
+export default class SoundAnalyser {
+    webAudioContextInstance: any;
+    mediaDevicesInstance: any;
+    deviceInstance: any;
+    deviceList: Array<string>;
+    context: any;
+    soundSamples: { clap : Array<number> };
+
+    constructor(webAudioContextInstance: any, mediaDevicesInstance: any, soundSamples:  { clap : Array<number> }) {
+        this.webAudioContextInstance = webAudioContextInstance;
+        this.mediaDevicesInstance = mediaDevicesInstance;
+
+        this.deviceList = [];
+        this.context = new webAudioContextInstance();
+        this.soundSamples = soundSamples;
+    }
+
+    getMicrophones = () => {
```

Comment:
This method records the microphones in an array `deviceList` but that array is never used. I think we should instead return a Promise with the list of microphone ids.

Reply:
Thinking about the name here. `getAudioInputDevices` might be a better name.